### PR TITLE
Add Ivory car pack support

### LINKED ISSUE #1263: Integrate Ivory Car Pack (civ cars)
This pack would allow for a large amount of variety in the civilian vehicles in the game. 

Would also be a place to try making additive civ templates, i.e. do not replace existing entries, while keeping loading separate.

### PR DESCRIPTION
**Pull request by SonorousMeerkat**
## What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement

### What have you changed and why?
Information:
    Worked out how to create an additive civ template,
    Created such a template for Ivory Car Pack,
    Added detection for said pack,
    Added extra logging to the template selection for verifying what loads. (in case there are future issues).

### Please specify which Issue this PR Resolves.
closes #1263 

### Please verify the following and ensure all checks are completed.

1. [ ] Have you loaded the mission in singleplayer?
2. [x] Have you loaded the mission in LAN host?
3. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 
  Load into any modset/map with the car pack in your mod list. Check for civilian spawns.
  Optionally, you can look at the output from monitoring civVehiclesWeighted in the debug console for any classnames with ivory_ at the start.
********************************************************
Notes:
  Super/Hypercars are exceedingly rare, during testing i found 1 in the wild. Does make the game look a bit like arma life :)

### CODE REVIEW

**Comment by SonorousMeerkat** [comment]
https://steamcommunity.com/sharedfiles/filedetails/?id=1888644057

**Comment by Lazejun** [comment]
I mean i agree that its different and something weird too.

But AAAAAAAAAA 0 Cargo Capable vehicles in Civ Pool D:

But who cares :D

**Comment by SonorousMeerkat** [comment]
Not true, this adds these vehicles to the existing set from the modset.
Trucks etc. still spawn.

File: A3-Antistasi/Templates/selector.sqf [hunk 1]
```diff
@@ -2,33 +2,36 @@
     This file controls the selection of templates based on the mods loaded and map used.
     When porting new mods/maps be sure to add them to their respective sections!
 */
-
+_filename = "selector.sqf";
```
[the remark this answers is not in the document]
Comment (SonorousMeerkat) [reply A3-Antistasi/Templates/selector.sqf]: done

File: A3-Antistasi/functions/init/fn_initVarServer.sqf [hunk 2]
```diff
@@ -446,7 +446,7 @@ private _templateVariables = [
 } forEach _templateVariables;
 
 call compile preProcessFileLineNumbers "Templates\selector.sqf";
-
+_fileName = "initVarServer.sqf"; //reset filename to un-confuse the logs
```
[the remark this answers is not in the document]
Comment (SonorousMeerkat) [reply A3-Antistasi/functions/init/fn_initVarServer.sqf]: removed

File: A3-Antistasi/Templates/Ivory_Civ.sqf [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	,"ivory_cv",1.0					//Ford Crown Victoria
+	,"ivory_190e",1.0				//Mercedes Benz 190E
+	,"ivory_wrx",1.0				//Subaru Impretza WRX
+	,"ivory_evox",1.0				//Mitsubishi Evo X
+	,"ivory_lfa",0.5				//Lexus LFA
+	,"ivory_supra",0.5			//Toyota Supra
+	,"ivory_gt500",0.5			//(Shelby) Ford Mustang GT500
+	,"ivory_r34",0.5				//Nissan Skyline R34
+	,"ivory_challenger",0.5	//Dodge Challenger
+	,"ivory_charger",0.5		//Dodge Charger
```
[the remark this answers is not in the document]
Comment (SonorousMeerkat) [reply A3-Antistasi/Templates/Ivory_Civ.sqf:19]: set to increase variety in line with the vehicle type/performance.

As it stands, they barely ever spawn below 1.0

File: A3-Antistasi/Templates/selector.sqf [hunk 4]
```diff
@@ -114,24 +129,31 @@ if(teamplayer != independent) then {//This section is for Altis Blufor ONLY!
...
+          [2, "Using arid FIA, NATO, CSAT, Vanilla Civ Templates", _filename] call A3A_fnc_log;
         };
       };
       call compile preProcessFileLineNumbers "Templates\Vanilla_Civ.sqf";
     };
   };
 };
+
+//Addon mod loading
+if (hasIvory) then {call compile preProcessFileLineNumbers "Templates\Ivory_Civ.sqf";[2, "Using Addon Ivory Cars Template", _filename] call A3A_fnc_log;};
```
[the remark this answers is not in the document]
Comment (SonorousMeerkat) [reply A3-Antistasi/Templates/selector.sqf]: Done

**Comment by SonorousMeerkat** [comment]
Replaced by #1283

**Pull request closed without merging**